java/src/jmri/jmrix/can/adapters/gridconnect/GcSerialDriverAdapter.java

```diff
@@ -128,14 +133,155 @@ protected GcTrafficController makeGcTrafficController() {
         return new GcTrafficController();
     }
 
+    /**
+     * Helper class wrapping the input serial port's InputStream. It starts a helper thread at
+     * high priority that reads the input serial port as fast as it can, buffering all incoming
+     * data in memory in a queue. The queue in unbounded and readers will get the data from the
+     * queue.
+     * <p>
+     * This class is thread-safe.
+     */
+    private class AsyncBufferInputStream extends FilterInputStream {
+        AsyncBufferInputStream(InputStream inputStream, String portName) {
+            super(inputStream);
+            Thread rt = new Thread(this::readThreadBody);
+            rt.setName("GcSerialPort InputBufferThread " + portName);
+            rt.setDaemon(true);
+            rt.setPriority(Thread.MAX_PRIORITY);
+            rt.start();
+        }
+
+        /**
+         * Helper function that tries to perform a read from the underlying port with a given
+         * maximum length.
+         * @param len how many bytes to request from the port. Setting this to 1 will apparently
+         *            block the thread if there are zero bytes available.
+         * @return a block of data read, or nullptr if fatal IO errors make further use
+         * of this port impossible.
+         */
+        private BufferEntry tryRead(int len) {
+            BufferEntry tail = new BufferEntry();
+            try {
+                tail.data = new byte[len];
+                tail.len = in.read(tail.data, 0, len);
+                errorCount = 0;
+            } catch (IOException e) {
+                tail.e = e;
+                e.printStackTrace();
```

Comment:
Do not use printStackTrace, but include the exception in the logging statement. See the SLF4J JavaDocs for details.

Reply:
done